improver_tests/blending/weighted_blend/test_MergeCubesForWeightedBlending.py

```diff
@@ -387,6 +388,32 @@ def test_blend_realizations(self):
         self.assertEqual(result[0].metadata, cube1.metadata)
         self.assertEqual(result[1].metadata, cube2.metadata)
 
+    def test_handling_blend_time(self):
+        """Test merging works with mismatched and / or missing blend time
+        coordinates"""
+        cube_nowcast = self.cube_ukv.copy()
+        cube_nowcast.coord("forecast_reference_time").points = (
+            cube_nowcast.coord("forecast_reference_time").points[0] + 900
+        )
+        cube_nowcast.coord("forecast_period").points = (
+            cube_nowcast.coord("forecast_period").points[0] - 900
+        )
+        cube_nowcast.attributes["mosg__model_configuration"] = "nc_det"
+
+        blt_ukv = self.cube_ukv.coord("forecast_reference_time").copy()
```

Comment:
Ah! The ole Bacon, Lettuce and Tomato!
This first use of `blt` is confusing as it doesn't mention blend_time. You might be able to change it to use `.copy(name="blend_time")` so that this is a bit clearer. Or just use an expanded variable name: `blend_time_ukv`. These three statements are sufficiently simple that I think you could combine them into one line and avoid having that variable at all.

Reply:
No I can't, as iris does renaming in place rather than returning a copy.  So I do need this variable.  But I can remove the bacon as needed!

Reply:
I had thought you could do the rename as part of the cube.copy function, but you can only change the data there. Thanks for changing the variable name.